### ISSUE TITLE
damidseq_pipeline Killed

### ISSUE THREAD

**Issue by ademcan** [owenjm/damidseq_pipeline#5]
Hi Owen,
I am having some issues with damidseq_pipeline while analyzing a new dataset. I am using the exact same command line (with the correct input bam files) as previously, but the program always ends with a Killed signal. The same command works with other datasets. The only difference is that this time I have a single-end dataset, compared to paire-end previously. The program freezes at "Generating bins from..." and then is stuck at 0.0 % before getting Killed after a few hours.
Did you already have such issues previously? Thank you for your help.

I would also like to take the opportunity to wish you a Happy New Year :)

**Comment by ademcan** [comment]
It seems that there is something wrong with the following declaration
$cigar =~ s/[^\d]//g;
In this case, if the CIGAR is 87M1I2M1I6M, the value of $cigar is 871216, and then it tries to add this value to $pos such as 
$pend = $pos+$cigar;
Do you think that this is the issue here ?
Thanks.

**Comment by owenjm** [comment]
Hi Adem,
Happy 2017 to you too!  Sorry to hear about the issue; can I check that you were using v1.4.3 (I fixed a bug with chromosome name incompatibilities that could potentially cause this).
You're right that the CIGAR string handling will cause problems, though -- that code should have been changed when I brought in BAM file compatibility from other sources.  The correct code should be (as per the SAM handling routines):
```
foreach my $cig ($cigar =~ m/(\d+)[M|I|S|=|X]/g) {
	$readlen += $cig;
}
$pstart = $pos;
$pend = $pos+$readlen;
```
I'll send you an email with the modified file to test.

Cheers,
Owen
